core/src/test/java/org/elasticsearch/node/NodeTests.java

```diff
@@ -241,6 +242,34 @@ public void testDefaultPathDataNotSet() throws IOException {
         }
     }
 
+    public void testTribeNodeDeprecation() throws IOException {
```

Comment:
Does this test need to be so complicated, can it just be a test in `TribeServiceTests` where we simply try to construct a `TribeService` with and without a tribe node enabled?